PickList causes ClassCastException when resetting values

### Describe the bug

We are getting this exception:

```
jakarta.faces.convert.ConverterException: java.lang.ClassCastException: class java.lang.String cannot be cast to class [[Ljava.lang.String; (java.lang.String and [[Ljava.lang.String; are in module java.base of loader 'bootstrap')
	at org.primefaces.component.picklist.PickListRenderer.getConvertedValue(PickListRenderer.java:353)
	at org.primefaces.component.picklist.PickListRenderer.getModelValueToRender(PickListRenderer.java:414)
	at org.primefaces.component.picklist.PickListRenderer.encodeMarkup(PickListRenderer.java:75)
	at org.primefaces.component.picklist.PickListRenderer.encodeEnd(PickListRenderer.java:68)
	at jakarta.faces.component.UIComponentBase.encodeEnd(UIComponentBase.java:586)
	at jakarta.faces.component.UIComponent.encodeAll(UIComponent.java:1443)
	at jakarta.faces.render.Renderer.encodeChildren(Renderer.java:146)
	at jakarta.faces.component.UIComponentBase.encodeChildren(UIComponentBase.java:557)
	at jakarta.faces.component.UIComponent.encodeAll(UIComponent.java:1436)
	at com.sun.faces.context.PartialViewContextImpl$PhaseAwareVisitCallback.visit(PartialViewContextImpl.java:624)
	at com.sun.faces.component.visit.PartialVisitContext.invokeVisitCallback(PartialVisitContext.java:158)
	at jakarta.faces.component.UIForm.visitTree(UIForm.java:354)
	at jakarta.faces.component.UIComponent.visitTree(UIComponent.java:1279)
	at jakarta.faces.component.UIComponent.visitTree(UIComponent.java:1279)
	at jakarta.faces.component.UIComponent.visitTree(UIComponent.java:1279)
	at com.sun.faces.context.PartialViewContextImpl.processComponents(PartialViewContextImpl.java:394)
	at com.sun.faces.context.PartialViewContextImpl.processPartial(PartialViewContextImpl.java:312)
	at org.primefaces.context.PrimePartialViewContext.processPartial(PrimePartialViewContext.java:61)
	at jakarta.faces.context.PartialViewContextWrapper.processPartial(PartialViewContextWrapper.java:239)
	at org.omnifaces.context.OmniPartialViewContext.processPartial(OmniPartialViewContext.java:122)
	at jakarta.faces.component.UIViewRoot.encodeChildren(UIViewRoot.java:1092)
	at jakarta.faces.component.UIComponent.encodeAll(UIComponent.java:1436)
	at com.sun.faces.application.view.FaceletViewHandlingStrategy.renderView(FaceletViewHandlingStrategy.java:422)
	at com.sun.faces.application.view.MultiViewHandler.renderView(MultiViewHandler.java:160)
	at jakarta.faces.application.ViewHandlerWrapper.renderView(ViewHandlerWrapper.java:125)
	at jakarta.faces.application.ViewHandlerWrapper.renderView(ViewHandlerWrapper.java:125)
	at jakarta.faces.application.ViewHandlerWrapper.renderView(ViewHandlerWrapper.java:125)
	at org.omnifaces.viewhandler.OmniViewHandler.renderView(OmniViewHandler.java:166)
	at jakarta.faces.application.ViewHandlerWrapper.renderView(ViewHandlerWrapper.java:125)
	at com.sun.faces.lifecycle.RenderResponsePhase.execute(RenderResponsePhase.java:93)
	at com.sun.faces.lifecycle.Phase.doPhase(Phase.java:72)
	at com.sun.faces.lifecycle.LifecycleImpl.render(LifecycleImpl.java:150)
	at jakarta.faces.webapp.FacesServlet.executeLifecyle(FacesServlet.java:690)
	at jakarta.faces.webapp.FacesServlet.service(FacesServlet.java:449)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:195)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:140)
	at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:51)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:164)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:140)
	at org.ocpsoft.rewrite.servlet.RewriteFilter.doFilter(RewriteFilter.java:226)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:164)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:140)
	at org.omnifaces.facesviews.FacesViewsForwardingFilter.filterExtensionLessToExtension(FacesViewsForwardingFilter.java:187)
	at org.omnifaces.facesviews.FacesViewsForwardingFilter.filterExtensionLess(FacesViewsForwardingFilter.java:144)
	at org.omnifaces.facesviews.FacesViewsForwardingFilter.filterExtensionLess(FacesViewsForwardingFilter.java:130)
	at org.omnifaces.facesviews.FacesViewsForwardingFilter.doFilter(FacesViewsForwardingFilter.java:86)
	at org.omnifaces.filter.HttpFilter.doFilter(HttpFilter.java:108)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:164)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:140)
	at org.primefaces.webapp.filter.NoCacheFilter.doFilter(NoCacheFilter.java:45)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:164)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:140)
	at org.apache.catalina.core.ApplicationDispatcher.invoke(ApplicationDispatcher.java:632)
	at org.apache.catalina.core.ApplicationDispatcher.processRequest(ApplicationDispatcher.java:408)
	at org.apache.catalina.core.ApplicationDispatcher.doForward(ApplicationDispatcher.java:330)
	at org.apache.catalina.core.ApplicationDispatcher.forward(ApplicationDispatcher.java:267)
	at org.ocpsoft.rewrite.servlet.impl.HttpRewriteResultHandler.handleResult(HttpRewriteResultHandler.java:42)
	at org.ocpsoft.rewrite.servlet.RewriteFilter.rewrite(RewriteFilter.java:297)
	at org.ocpsoft.rewrite.servlet.RewriteFilter.doFilter(RewriteFilter.java:198)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:164)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:140)

Caused by: java.lang.ClassCastException: class java.lang.String cannot be cast to class [[Ljava.lang.String; (java.lang.String and [[Ljava.lang.String; are in module java.base of loader 'bootstrap')
	at org.primefaces.component.picklist.PickListRenderer.getConvertedValue(PickListRenderer.java:342)
	... 92 more
```

Following these steps with the reproducer code:

1. Click on "Save" (as the picklist has required="true" we get the error saying that we haven't selected a value).
2. Click on "Clean".

The problem is that in the method PickListRenderer#getConvertedValue, the parameter submittedValue is taking the value "" (empty string), and of course it cannot convert it to String[][]:

```
public Object getConvertedValue(FacesContext context, UIComponent component, Object submittedValue) throws ConverterException {
        try {
            PickList pickList = (PickList) component;
            String[][] value = (String[][]) submittedValue;
            String[] sourceValue = value[0];
            String[] targetValue = value[1];
            DualListModel model = new DualListModel();

            pickList.populateModel(context, sourceValue, model.getSource());
            pickList.populateModel(context, targetValue, model.getTarget());

            return model;
        }
        catch (Exception exception) {
            throw new ConverterException(exception);
        }
    }
```

### Reproducer

XHTML:

```
<h:form>
    <p:pickList value="#{bean.picklist}" var="item" itemValue="#{item}" itemLabel="#{item}" required="true"/>
    <h:commandButton value="Save">
	<p:ajax process="@form" update="@form" listener="#{bean.save}" />
    </h:commandButton>
    <h:commandButton value="Clean">
	<p:ajax update="@form" listener="#{bean.clean}" resetValues="true"/>
    </h:commandButton>
</h:form>
```

BEAN:

```
private DualListModel<String> picklist;

public void save() { }

public void clean() { }

public DualListModel<String> getPicklist() {
	if (this.picklist== null) {
		final List<String> list = new ArrayList<>();
		list.add("1");
		list.add("2");
		this.picklist= new DualListModel<>(list, new ArrayList<>());
	}

	return this.picklist;
}

public void setPicklist(DualListModel<String> picklist) {
	this.picklist= picklist;
}
```

### Expected behavior

No exception

### PrimeFaces edition

None

### PrimeFaces version

14.0.5

### Theme

_No response_

### JSF implementation

Mojarra

### JSF version

4.1.0

### Java version

17

### Browser(s)

Chrome 129

--- a/primefaces/src/main/java/org/primefaces/component/picklist/PickListRenderer.java
+++ b/primefaces/src/main/java/org/primefaces/component/picklist/PickListRenderer.java
@@ -414,7 +414,7 @@ public class PickListRenderer extends InputRenderer {
 
     protected DualListModel getModelValueToRender(FacesContext context, PickList pickList) {
         Object submittedValue = pickList.getSubmittedValue();
-        if (submittedValue != null) {
+        if (submittedValue != null && submittedValue instanceof String[][]) {
             return (DualListModel) getConvertedValue(context, pickList, submittedValue);
         }
 

